fs/io_zenfs.cc

```diff
@@ -705,6 +737,38 @@ void ZoneFile::ReplaceExtentList(std::vector<ZoneExtent*> new_list) {
   extents_ = new_list;
 }
 
+IOStatus ZoneFile::SetLinkInfo(std::string linkf) {
+  WriteLock lck(this);
```

Comment:
We're holding the files_ mutex, why do we need this lock?

Reply:
Yes, we do not this, holding the files_ mutex should be sufficient. Removed it.